source/openpulse/openpulse/parser.py

```diff
@@ -0,0 +1,303 @@
+OpenPulseNodeVisitor.visitEndStatement = QASMNodeVisitor.visitEndStatement
+OpenPulseNodeVisitor.visitEqualityExpression = QASMNodeVisitor.visitEqualityExpression
+OpenPulseNodeVisitor.visitExpressionStatement = QASMNodeVisitor.visitExpressionStatement
+OpenPulseNodeVisitor.visitExternArgument = QASMNodeVisitor.visitExternArgument
+OpenPulseNodeVisitor.visitExternStatement = QASMNodeVisitor.visitExternStatement
+OpenPulseNodeVisitor.visitForStatement = QASMNodeVisitor.visitForStatement
+OpenPulseNodeVisitor.visitGateCallStatement = QASMNodeVisitor.visitGateCallStatement
+OpenPulseNodeVisitor.visitGateModifier = QASMNodeVisitor.visitGateModifier
+OpenPulseNodeVisitor.visitGateOperand = QASMNodeVisitor.visitGateOperand
+OpenPulseNodeVisitor.visitGateStatement = QASMNodeVisitor.visitGateStatement
+OpenPulseNodeVisitor.visitIfStatement = QASMNodeVisitor.visitIfStatement
+OpenPulseNodeVisitor.visitIncludeStatement = QASMNodeVisitor.visitIncludeStatement
+OpenPulseNodeVisitor.visitIndexExpression = QASMNodeVisitor.visitIndexExpression
+OpenPulseNodeVisitor.visitIndexedIdentifier = QASMNodeVisitor.visitIndexedIdentifier
+OpenPulseNodeVisitor.visitIoDeclarationStatement = QASMNodeVisitor.visitIoDeclarationStatement
+OpenPulseNodeVisitor.visitLiteralExpression = QASMNodeVisitor.visitLiteralExpression
+OpenPulseNodeVisitor.visitLogicalAndExpression = QASMNodeVisitor.visitLogicalAndExpression
+OpenPulseNodeVisitor.visitLogicalOrExpression = QASMNodeVisitor.visitLogicalOrExpression
+OpenPulseNodeVisitor.visitMeasureArrowAssignmentStatement = (
+    QASMNodeVisitor.visitMeasureArrowAssignmentStatement
+)
+OpenPulseNodeVisitor.visitMeasureExpression = QASMNodeVisitor.visitMeasureExpression
+OpenPulseNodeVisitor.visitMultiplicativeExpression = QASMNodeVisitor.visitMultiplicativeExpression
+OpenPulseNodeVisitor.visitOldStyleDeclarationStatement = (
+    QASMNodeVisitor.visitOldStyleDeclarationStatement
+)
+OpenPulseNodeVisitor.visitParenthesisExpression = QASMNodeVisitor.visitParenthesisExpression
+OpenPulseNodeVisitor.visitPowerExpression = QASMNodeVisitor.visitPowerExpression
+OpenPulseNodeVisitor.visitPragma = QASMNodeVisitor.visitPragma
+OpenPulseNodeVisitor.visitProgram = QASMNodeVisitor.visitProgram
+OpenPulseNodeVisitor.visitQuantumDeclarationStatement = (
+    QASMNodeVisitor.visitQuantumDeclarationStatement
+)
+OpenPulseNodeVisitor.visitResetStatement = QASMNodeVisitor.visitResetStatement
+OpenPulseNodeVisitor.visitScope = QASMNodeVisitor.visitScope
+OpenPulseNodeVisitor.visitSetExpression = QASMNodeVisitor.visitSetExpression
+OpenPulseNodeVisitor.visitStatement = QASMNodeVisitor.visitStatement
+OpenPulseNodeVisitor.visitStatementOrScope = QASMNodeVisitor.visitStatementOrScope
+OpenPulseNodeVisitor.visitUnaryExpression = QASMNodeVisitor.visitUnaryExpression
+OpenPulseNodeVisitor.visitWhileStatement = QASMNodeVisitor.visitWhileStatement
+
+
+class CalParser(QASMVisitor[None]):
```

Comment:
Before [delegating-parser](https://github.com/jakelishman/openqasm/tree/delegating-parser) is merged into OpenQASM mainline, I used this Visitor to simulate the effect.